composer.json

```diff
@@ -77,5 +77,59 @@
             "ergebnis/composer-normalize": true
         },
         "sort-packages": true
+    },
+    "scripts": {
+        "cs:check": "php php-cs-fixer fix --dry-run",
+        "cs:diff": "git diff HEAD --name-only --diff-filter=ACMR -- '*.php' | xargs -r php php-cs-fixer fix --dry-run --diff",
+        "cs:diff-fix": "git diff HEAD --name-only --diff-filter=ACMR -- '*.php' | xargs -r php php-cs-fixer fix",
+        "cs:fix": "php php-cs-fixer fix",
+        "dev-tools:check": [
+            "./dev-tools/check_file_permissions.sh",
+            "./dev-tools/check_trailing_spaces.sh"
+        ],
+        "dev-tools:docs": "php dev-tools/doc.php",
+        "dev-tools:install": "composer -d dev-tools install",
+        "mess-detector": "php dev-tools/vendor/bin/phpmd . ansi phpmd.xml",
+        "phpstan": "php -d memory_limit=256M dev-tools/vendor/bin/phpstan analyse",
+        "qa": "@quality-assurance",
+        "quality-assurance": [
+            "Composer\\Config::disableProcessTimeout",
+            "@cs:check",
+            "@sa",
+            "@test"
+        ],
+        "sa": "@static-analysis",
+        "static-analysis": "@phpstan",
```

Comment:
let's call it phpstan.
we call ourself to be static-analysis tool as well, so...

Reply:
There's `phpstan` script too, this is only an alias which can be converted into a group of script if needed. So either we can get rid of `static-analysis` / `sa` scripts altogether, or include Fixer here and reorganise `qa` script to run `sa` and `test`.

I would go with 1st option.

Reply:
I added `cs:check` to `static-analysis`. I decided to use `cs:check`, not `cs:fix` to be consistent with `phpstan` - it also only reports problems, doesn't fix them. So is `sa` script exits with non-0 code, it requires manual action (fixing PHPStan errors or running `cs:fix`).